promdump/promdump.go

```diff
@@ -651,9 +770,51 @@ func main() {
 		}
 	}
 	if *metric != "" {
-		err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
+		customMetricCount, err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
 		if err != nil {
 			log.Fatalln("exportMetric:", err)
 		}
 	}
+
+	if *enableTar {
+		if *tarFilename == "" {
+			// If filename is not provided, generate a default tar filename
+			*tarFilename = generateDefaultTarFilename()
+		} else {
+			// Check if filename already exists
+			if _, err := os.Stat(*tarFilename); err == nil {
+				log.Fatalf("specified --tar_filename '%s' already exists; please choose a different filename", *tarFilename)
+				return
+			}
+		}
+
+		TarFileOut, err := os.Create(*tarFilename)
```

Comment:
Case matters for Golang variables. Starting a variable with a capital letter tells Golang that the variable should be a public member (i.e. it will be available to external callers). Change to `tarFileOut` since we're currently not exporting anything from this module (on purpose).

Reply:
Change TarFileOut to tarFileOut